vignettes/corrected-se.Rmd

```diff
@@ -0,0 +1,311 @@
+
+First-order correction for SE
+
+$$\hat V_{\gamma, c} = \hat V_{\gamma} + (\boldsymbol{H}_\gamma)^{-1} \left(\frac{\partial^2 \ell}{\partial \theta \partial \gamma^\top}\right) \hat V_{\theta} \left(\frac{\partial^2 \ell}{\partial \theta \partial \gamma^\top}\right)^\top (\boldsymbol{H}_\gamma)^{-1},$$
+
+where $V_{\gamma}$ is the naive covariance matrix of the structural parameter estimates $\hat{\boldsymbol{\gamma}}$ assuming the measurement error variance parameter, $\boldsymbol{\theta}$, is known, $\boldsymbol{H}_\gamma$ is the Hessian matrix of the log-likelihood $\ell$ with respect to $\hat{\boldsymbol{\gamma}}$, and $V_{\theta}$ can be obtained in the first-stage measurement model analysis.
+
+```{r load-pkg, message = FALSE}
+library(lavaan)
+library(R2spa)
+library(boot)
+```
+
+This example is based on the Political Democracy data used in <https://lavaan.ugent.be/tutorial/sem.html> and as an example in Bollen (1989)'s book.
+
+## Separate Measurement Models
+
+```{r}
+model <- '
+  # latent variable definitions
+     ind60 =~ x1 + x2 + x3
+     dem60 =~ y1 + y2 + y3 + y4
+
+  # regressions
+    dem60 ~ ind60
+'
+```
+
+```{r}
+cfa_ind60 <- cfa("ind60 =~ x1 + x2 + x3", data = PoliticalDemocracy)
+# Regression factor scores
+fs1 <- get_fs_lavaan(cfa_ind60, vfsLT = TRUE)
+# Delta method variance of (loading, intercept)
+vldev1 <- attr(fs1, which = "vfsLT")
+cfa_dem60 <- cfa("dem60 =~ y1 + y2 + y3 + y4",
+                 data = PoliticalDemocracy)
+# Regression factor scores
+fs2 <- get_fs_lavaan(cfa_dem60, vfsLT = TRUE)
+# Delta method variance of (loading, intercept)
+vldev2 <- attr(fs2, which = "vfsLT")
+fs_dat <- data.frame(
+  fs_ind60 = fs1$fs_ind60,
+  fs_dem60 = fs2$fs_dem60
+)
+# Combine sampling variance of loading and error variance
+# Note: loadings first, then error variance
+vldev <- block_diag(vldev1, vldev2)[c(1, 3, 2, 4), c(1, 3, 2, 4)]
+```
+
+```{r}
+# 2S-PA
+# Assemble loadings
+ld <- block_diag(attr(fs1, which = "fsL"),
+                 attr(fs2, which = "fsL"))
+ev <- block_diag(attr(fs1, which = "fsT"),
+                 attr(fs2, which = "fsT"))
+tspa_fit <- tspa(model = "dem60 ~ ind60",
+                 data = fs_dat,
+                 cross_loadings = ld,
+                 vc = ev)
+# Unadjusted covariance
+vcov(tspa_fit)
+# Adjusted covariance matrix
+(vc_cor <- vcov_corrected(tspa_fit, vfsLT = vldev, which_free = c(1, 4, 5, 7)))
+# Corrected standard errors
+sqrt(diag(vc_cor))
+```
+
+Standardized solution
+
+```{r, eval = FALSE}
+tspa_est_std <- function(se) {
+  tfit <- tspa(model = "dem60 ~ ind60",
+               data = fs_dat,
+               se = list(ind60 = se[1], dem60 = se[2]))
+  standardizedSolution(tfit)$est.std[5]
+}
+tspa_est_std(unlist(se_fs))
```

Comment:
`se_fs` is not defined so this part is not executable. I noticed that you put `eval = FALSE` and guess you may just want to show the syntax. In case you want people to run this code, `se_fs` may need to be defined earlier.

Reply:
Thanks for checking. I was deciding whether to include this part or not. I guess it might help to include this, and will push an update that works in a second.